### PR TITLE
feat(sdk): allow photon to force mount a sub FastAPI as asgi proxy, and allow overloading openapi function

### PR DESCRIPTION
**Pull request by Yangqing**

### CODE REVIEW

File: leptonai/photon/photon.py [hunk 1]
```diff
@@ -892,12 +906,18 @@ def _mount_route(self, app, path, func):
             has_flask = False
 
         if isinstance(subapp, FastAPI):
-            app.include_router(subapp.router, prefix=f"/{path}")
+            if force_asgi_proxy:
```
Comment (bddppq) [inline leptonai/photon/photon.py:909]: I think this is unrelated to asgi_proxy, maybe call it `mount_method="mount"` or `mount_method="include_router"`?

**Review by bddppq — APPROVED** [review]

File: leptonai/photon/photon.py (hunk 1 above)
Comment (Yangqing) [reply leptonai/photon/photon.py:909]: My fault, haven't completely updated the diff yet... let me update

Comment (Yangqing) [reply leptonai/photon/photon.py]: Changed this to "use_router_if_possible=True", since this only applies to when the subapp is a fastapi app.
